### PR TITLE
add ovs forwarder deployment examples

### PR DESCRIPTION
**Pull request by pperiyasamy**
adds ovs forwarder deployment examples, would help to test [cmd-forwarder-ovs](https://github.com/networkservicemesh/cmd-forwarder-ovs) for which the code development is on going.

Depends on cmd-forwarder-ovs [PR]( https://github.com/networkservicemesh/cmd-forwarder-ovs/pull/7).

Signed-off-by: Periyasamy Palanisamy <periyasamy.palanisamy@est.tech>

### CODE REVIEW

**Review by Mixaster995 — APPROVED** [review]

File: examples/ovs/README.md
Comment (d-uzlov) [inline examples/ovs/README.md:1]:
```suggestion
# OVS examples
```

**Review by d-uzlov — APPROVED** [review]

File: examples/use-cases/SmartVF2SmartVF/README.md
Comment (Bolodya1997) [inline examples/use-cases/SmartVF2SmartVF/README.md:63]: We have only `10G` NICs on our packet integration clusters, so this client will never start on them. Please use here `10G` instead of `100G`.

File: examples/use-cases/SmartVF2SmartVF/README.md
Comment (Bolodya1997) [inline examples/use-cases/SmartVF2SmartVF/README.md:1]: What is the difference with the `Kernel2Kernel` use case?

Comment (Bolodya1997) [reply examples/use-cases/SmartVF2SmartVF/README.md:1]: I mean instead of name :)

Comment (pperiyasamy) [reply examples/use-cases/SmartVF2SmartVF/README.md:1]: In SmartVF2SmartVF case, both Client and Endpoint pod are attached with smart VF (on top of this [PR](https://github.com/networkservicemesh/sdk-sriov/pull/179), but design approach for End point connection is under question (i.e. one VF per client connection) and discussing other possibilities [here](https://docs.google.com/document/d/10PC-291O4bKfYFQxVZnFquVcK8p5wDWchnAdxkODCLk/edit?usp=sharing)).

**Comment by Bolodya1997** [comment]
LGTM, but we actually are not able to run this on our packet integration cluster, because we lack NIC needed for it. So there possibly can be problems with validating these new examples after some future NSM updates.

@edwarnicke @denis-tingaikin 
Thoughts?

File: examples/use-cases/SmartVF2SmartVF/README.md
Comment (edwarnicke) [reply examples/use-cases/SmartVF2SmartVF/README.md:63]: @pperiyasamy  We should be able to CI the example provided... the packet.net env where we have HW to run such examples only has 10G nics... is it possible to get the example to conform to what we can actually test?  Feel free to talk in commentary for the example about how to alter it for MLX5 :)

File: apps/forwarder-host-ovs/forwarder.yaml
Comment (edwarnicke) [inline apps/forwarder-host-ovs/forwarder.yaml:20]: Typically we bind to particular version of a container (and they get updated from downstream).  Could that be done here?

File: apps/forwarder-ovs/forwarder.yaml
Comment (edwarnicke) [inline apps/forwarder-ovs/forwarder.yaml:20]: Typically we bind to particular version of a container (and they get updated from downstream).  Could that be done here?

File: examples/use-cases/SmartVF2SmartVF/README.md
Comment (pperiyasamy) [reply examples/use-cases/SmartVF2SmartVF/README.md:63]: @edwarnicke currently the plan is to configure the MLX5 100G NIC as per this [instruction](https://github.com/k8snetworkplumbingwg/ovs-cni/blob/main/docs/ovs-offload.md#mellanox-sr-iov-configuation) and create `pci.config` file and pass it as `NSM_SRIOV_CONFIG_FILE` into ovs forwarder. I think once this configuration is automated  (like its done for legacy sriov VFs with sdk-sriov pcifunction) by enhancing sdk-sriov, then this can be easily integrated with packet.net, WDYT ?

File: examples/use-cases/SmartVF2SmartVF/README.md
Comment (pperiyasamy) [reply examples/use-cases/SmartVF2SmartVF/README.md:1]: @Bolodya1997 I'm not sure if MLX provides 10G NIC which has hw offload capability, we've used only 100G Connectx-5 NIC for testing. anyway this NIC is totally different from legacy SRIOV NIC and also different drivers to be bound on PFs and VFs. we can still use same sriov.conf file, but this would go into another `physicalFunctions`.

File: apps/forwarder-host-ovs/forwarder.yaml
Comment (edwarnicke) [reply apps/forwarder-host-ovs/forwarder.yaml:20]: @pperiyasamy Look at how we are [publishing docker containers in cmd-template](https://github.com/networkservicemesh/cmd-template/blob/main/.github/workflows/docker-push-ghcr.yml).  Its appears you already have that, so all you'd need to do here is use your latest [githash](https://github.com/networkservicemesh/cmd-forwarder-ovs/pkgs/container/ci%2Fcmd-forwarder-ovs)  (currently 811ba56 I believe).  After that [update deployments](https://github.com/networkservicemesh/cmd-forwarder-ovs/blob/main/.github/workflows/update-deployments.yaml) should kick in and maintain it for you :)

Comment (pperiyasamy) [reply apps/forwarder-host-ovs/forwarder.yaml:20]: @edwarnicke ah! ok, that's awesome. Thanks. Let me fix CI for [Dockerfile.use-host-ovs](https://github.com/networkservicemesh/cmd-forwarder-ovs/blob/main/Dockerfile.use-host-ovs) in `cmd-forwarder-ovs `repo and then will update this PR.

Comment (pperiyasamy) [reply apps/forwarder-host-ovs/forwarder.yaml]: done. The [PR]( https://github.com/networkservicemesh/cmd-forwarder-ovs/pull/7) is also raised to address this so that update deployments would kick in automatically to change image version.

File: apps/forwarder-ovs/forwarder.yaml
Comment (pperiyasamy) [reply apps/forwarder-ovs/forwarder.yaml]: done as above.

**Review by Bolodya1997 — APPROVED** [review]

**Review by glazychev-art — APPROVED** [review]

**Comment by denis-tingaikin** [comment]
@pperiyasamy Did we merge all required PRs?

**Comment by pperiyasamy** [comment]
> @pperiyasamy Did we merge all required PRs?

@denis-tingaikin we need to merge the following PRs.
https://github.com/networkservicemesh/cmd-nse-icmp-responder/pull/286
https://github.com/networkservicemesh/sdk-sriov/pull/258

**Review by denis-tingaikin — APPROVED** [review]
Currently, we cannot merge this due we got a red build on kind related to the last merges.

Will merge this PR when we'll fix the problem.

**Pull request merged**
